File: task2.cpp
#include <iostream>
#include <random>
#include <vector>
#include <algorithm>
#include <unordered_map>

using namespace std;

void randomFill(vector<int>& fillVec, int N, int gr1, int gr2) {
    random_device rd;
    mt19937_64 random_(rd());
    for (int i = 0; i < N; i++) {
        fillVec.push_back(random_()%gr1 + gr2);
    }
}

void newMass(const vector<int>& rand1, const vector<int>& rand2, vector<int>& New) {
    int ind = 0;
    for (int n : rand1) {
        if (ind % 2 == 0) {
            New.push_back(n + rand2[ind]);
        }
        else {
            New.push_back(n - rand2[ind]);
        }
        ind++;
    }
}

int main() {
    //1-2
    int N = 10;
    vector<int> randNum;
    randomFill(randNum, N, 101, 100);
    sort(randNum.begin(), randNum.end());
    cout << "Массив с случайными значениями: ";
    for (int n : randNum) {
        cout << n << " ";
    }
    cout << endl;
    cout << "Второй по величине элемент: " << randNum[N-2] << endl;
    int sum = 0;
    randNum.pop_back();
    randNum.pop_back();
    randNum.erase(randNum.begin());
    for (int i : randNum) {
        sum += i;
    }
    cout << "Сумма: " << sum << endl;

    randNum.clear();
    //3
    vector<int> randNum1;
    vector<int> New;

    randomFill(randNum, N, 101, -50);
    randomFill(randNum1, N, 101, -50);

    newMass(randNum, randNum1, New);

    cout << "Первый массив с случайными значениями: ";
    for (int n : randNum) {
        cout << n << " ";
    }
    cout << endl;
    cout << "Второй массив с случайными значениями: ";
    for (int n : randNum1) {
        cout << n << " ";
    }
    cout << endl;
    cout << "Новый массив: ";
    for (int n : New) {
        cout << n << " ";
    }
    cout << endl;
    unordered_map<int, int> povtor;
    for (int n : New) {
        povtor[n]++;
    }
    for (pair<int, int> n : povtor) {
        cout << "Элемент: " << n.first << " Количество повторов: " << n.second << endl;
    }
    int a, b;
    cout << "Введите начальный и конечный год: ";
    cin >> a >> b;
    vector<int> visokYears;
    for (int i = a; i <= b; i++) {
        if (i%4 == 0) {
            visokYears.push_back(i);
        }
    }
    cout << "Високосные годы: ";
    for (int n : visokYears) {
        cout << n << " ";
    }
    return 0;
}

File: task2.cs
using System;
using System.Collections.Generic;

class Program
{
    static Random random_ = new Random();

    static void RandomFill(List<int> fillList, int N, int gr1, int gr2)
    {
        for (int i = 0; i < N; i++)
        {
            fillList.Add(random_.Next(gr2, gr1 + gr2));
        }
    }

    static void NewMass(List<int> rand1, List<int> rand2, List<int> New)
    {
        for (int i = 0; i < rand1.Count; i++)
        {
            if (i % 2 == 0)
            {
                New.Add(rand1[i] + rand2[i]);
            }
            else
            {
                New.Add(rand1[i] - rand2[i]);
            }
        }
    }

    static void Main(string[] args)
    {
        // 1-2
        int N = 10;
        List<int> randNum = new List<int>();
        RandomFill(randNum, N, 101, 100);
        randNum.Sort();
        Console.Write("Массив с случайными значениями: ");
        foreach (int n in randNum)
        {
            Console.Write(n + " ");
        }
        Console.WriteLine();
        Console.WriteLine("Второй по величине элемент: " + randNum[N - 2]);

        randNum.RemoveAt(N - 1);
        randNum.RemoveAt(N - 2);
        randNum.RemoveAt(0);

        int sum = 0;
        foreach (int i in randNum)
        {
            sum += i;
        }
        Console.WriteLine("Сумма: " + sum);

        randNum.Clear();
        // 3
        List<int> randNum1 = new List<int>();
        List<int> New = new List<int>();

        RandomFill(randNum, N, 101, -50);
        RandomFill(randNum1, N, 101, -50);

        NewMass(randNum, randNum1, New);

        Console.Write("Первый массив с случайными значениями: ");
        foreach (int n in randNum)
        {
            Console.Write(n + " ");
        }
        Console.WriteLine();
        Console.Write("Второй массив с случайными значениями: ");
        foreach (int n in randNum1)
        {
            Console.Write(n + " ");
        }
        Console.WriteLine();
        Console.Write("Новый массив: ");
        foreach (int n in New)
        {
            Console.Write(n + " ");
        }
        Console.WriteLine();

        Dictionary<int, int> povtor = new Dictionary<int, int>();
        foreach (int n in New)
        {
            if (povtor.ContainsKey(n))
            {
                povtor[n]++;
            }
            else
            {
                povtor.Add(n, 1);
            }
        }
        foreach (KeyValuePair<int, int> n in povtor)
        {
            Console.WriteLine("Элемент: " + n.Key + " Количество повторов: " + n.Value);
        }

        Console.Write("Введите начальный и конечный год: ");
        int a = Convert.ToInt32(Console.ReadLine());
        int b = Convert.ToInt32(Console.ReadLine());
        List<int> visokYears = new List<int>();
        for (int i = a; i <= b; i++)
        {
            if (i % 4 == 0)
            {
                visokYears.Add(i);
            }
        }
        Console.Write("Високосные годы: ");
        foreach (int n in visokYears)
        {
            Console.Write(n + " ");
        }
    }
}
